File: src/bog.js
const { Pool } = require('pg');
const { request } = require('express');
const pool = new Pool({
    user: 'hamzaasaad',
    host: 'localhost',
    database: 'hamzaasaad',
    password: null,
    port: 5431,
  })
pool.connect();

var time = new Date();
time.setDate(time.getDate() - 1);
console.log(time);


const update = 'update depositortest set firstrequesttime= $1, dailytime= $1, weeklytime= $1 where address= $2';
const values = [time,'0x066Adead2d82A1C2700b4B48ee82ec952b6b18dA'];
const result = pool.query(update,values);
pool.end();
console.log(result);

/*
async function confirm_transaction(addressDetails){   // make a column for unaccountedAmount in db
    let FAUCET_ADDRESS = ""
    let depositedTx = API.checkDeposit(addressDetails);
    let lastValidatedTx = addressDetails.validatedtx
    if (depositedTx){
      if (lastValidatedTx){
        for (let i; i < depositedTx.length; i++){
          if (depositedTx[i].amount == "32000000000000000000" && depositedTx[i].hash != lastValidatedTx && depositedTx[i].to == FAUCET_ADDRESS){
            addressDetails.weeklyCount += 33;
            addressDetails.dailyCount += 33;
            //update time in db
            addressDetails.validatedtx = depositedtx[i].hash
            //db.commit()
            return
          }else if (Number(depositedTx[i].amount) < 32000000000000000000 && depositedTx[i].to == FAUCET_ADDRESS){
            addressDetails.unaccountedAmount += Number(depositedTx[i].amount)
            addressDetails.unaccountedTx = depositedTx[i].hash
            //db.commit()
  
          }else if (Number(depositedTx[i].amount) > 32000000000000000000 && depositedTx[i].to == FAUCET_ADDRESS){
            addressDetails.unaccountedAmount += 32000000000000000000 - Number(depositedTx[i].amount)
            addressDetails.unaccountedTx = depositedTx[i].hash
            //db.commit()
          }
        }
      }else{
        for (let i; i < depositedTx.length; i++){
          if (depositedTx[i].amount == "32000000000000000000" && depositedTx[i].hash != last_validated_tx && depositedTx[i].to == FAUCET_ADDRESS){
            addressDetails.validatedtx = depositedtx[i].hash
            addressDetails.weeklyCount += 33;
            addressDetails.dailyCount += 33;
            //update time in db
            return
          }else if (Number(depositedTx[i].amount) < 32000000000000000000 && depositedTx[i].to == FAUCET_ADDRESS){
            addressDetails.unaccountedAmount += Number(depositedTx[i].amount)
            addressDetails.unaccountedTx = depositedTx[i].hash
            //db.commit()
  
          }else if (Number(depositedTx[i].amount) > 32000000000000000000 && depositedTx[i].to == FAUCET_ADDRESS){
            addressDetails.unaccountedAmount += 32000000000000000000 - Number(depositedTx[i].amount)
            addressDetails.unaccountedTx = depositedTx[i].hash
            //db.commit()
          }}
      }
      if (addressDetails.unaccountedAmount >= 32000000000000000000){
            addressDetails.unaccountedAmount -= 32000000000000000000
            addressDetails.weeklyCount += 33   
            addressDetails.dailyCount += 33
            //update time in db
            addressDetails.validatedTx = addressDetails.unaccountedTx
      }
    }else{
      return false;
    }
  }*/

File: src/bog.py
def confirm_transaction(address):
    deposited_tx = API.checkDeposit(address)
    last_validated_tx = get_validated_tx_from_db
    db_unaccounted_amount = f"SELECT UNACCOUTNED_AMOUNT WHERE ADDRESS={address}"
    deposit_complete = False

    if deposited_tx:
        if last_validated_tx:
          for i in range(len(deposited_tx)):
              if deposited_tx[i].amount == "32000000000000000000" and deposited_tx[i] != last_validated_tx and deposited_tx[i].to == FAUCET_ADDRESS:
                weeklyCount += 33   #update in db
                dailyCount += 33    #update in db
                # handle daily and weekly time in db
                last_validated_tx =  deposited_tx[i]  # update_last_validated_transaction_in_db
                return
              elif (int(deposited_tx[i].amount)) < 32000000000000000000 and deposited_tx[i].to == FAUCET_ADDRESS:
                db_unaccounted_amount += int(deposited_tx[i].amount)
                 # update unaccountedAmount and unaccountedTx in database
                f"UPDATE SET unaccoutned_amount where address={address}"
                f"UPDATE SET unaccoutned_tx where address={address}"
                return
        else:
          for i in range(len(deposited_tx)):
            if deposited_tx[i].amount == "32000000000000000000" and deposited_tx[i] != last_validated_tx and deposited_tx[i].to == FAUCET_ADDRESS:
                last_validated_tx = deposited_tx[i] # update_last_validated_transaction_in_db
                weeklyCount += 33   #update in db
                dailyCount += 33    #update in db
                # handle daily and weekly time in db
                return
            elif (int(deposited_tx[i].amount)) < 32000000000000000000 and deposited_tx[i].to == FAUCET_ADDRESS:
                db_unaccounted_amount += int(deposited_tx[i].amount)
                 # update unaccountedAmount and unaccountedTx in database
                f"UPDATE SET unaccoutned_amount where address={address}"
                f"UPDATE SET unaccoutned_tx where address={address}"
                return
        if db_unaccounted_amount >= 32000000000000000000:
          db_unaccounted_amount -= 32000000000000000000
          weeklyCount += 33   #update in db
          dailyCount += 33    #update in db
          # handle daily and weekly time in db
          last_validated_tx = unvalidated_tx   # update_last_validated_transaction_in_db
          
    else:
      return False